src/hub/action_request.ts

```diff
@@ -180,62 +180,41 @@ export class ActionRequest {
 
     const stream = new PassThrough()
     const returnPromise = callback(stream)
-    const timeout = process.env.ACTION_HUB_STREAM_REQUEST_TIMEOUT ?
-        parseInt(process.env.ACTION_HUB_STREAM_REQUEST_TIMEOUT, 10)
-      :
-        13 * 60 * 1000
 
     const url = this.scheduledPlan && this.scheduledPlan.downloadUrl
 
     const streamPromise = new Promise<void>((resolve, reject) => {
       if (url) {
         winston.info(`[stream] beginning stream via download url`, this.logInfo)
-        let hasResolved = false
-        httpRequest
-          .get(url, {timeout})
-          .on("error", (err) => {
-            if (hasResolved && (err as any).code === "ECONNRESET") {
-              winston.info(`[stream] ignoring ECONNRESET that occured after streaming finished`, this.logInfo)
+        try {
+          fetch(url).then((response: any) => {
+            if (response.ok) {
+              response.body.pipe(stream)
+                  .on("error", (err: any) => {
+                    winston.error(`[stream] PassThrough stream error`, this.logInfo)
+                    reject(err)
+                  })
+                  .on("finish", () => {
+                    winston.info(`[stream] PassThrough stream finished`, this.logInfo)
+                    resolve()
+                  })
+                  .on("close", () => {
+                    winston.info(`[stream] PassThrough stream closed`, this.logInfo)
```

Comment:
What should happen to the promise here?

Reply:
good catch, close is after finish for streaming events. We'll resolve() in the close event instead